src/components/data-pane/type-changer.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+
+import { ExpandedType } from 'compassql/build/src/query/expandedtype';
+import {DATASET_SCHEMA_CHANGE_FIELD_TYPE, DatasetSchemaChangeFieldType} from '../../actions/dataset';
+import {ActionHandler} from '../../actions/redux-action';
+import {ShelfFieldDef} from '../../models/shelf/encoding';
+import * as styles from './type-changer.scss';
+
+export interface TypeChangerProps extends ActionHandler<DatasetSchemaChangeFieldType> {
+  fieldDef: ShelfFieldDef;
```

Comment:
Instead of require `ShelfFieldDef` in which `field` can be either string or wildcard.  Let's not make `fieldDef` a prop and instead have `field: string` and `type` directly (since we're not using other parameters anyway.  :)